### PR TITLE
Pin requirements with pip-tools

### PR DESCRIPTION
**Pull request by Azd325**
Splits production and development packages in seperate requirements files.

In requirements.txt includes only the packages what are used for a
production enviroment. In requirements-dev.txt are production and development
dependencies unpinned to test on travis for the latest packages if something
breaks.

This commit adds also pip-tools as development tool to pin and sync the
requirements with the local enviroment.

Updates Makefile to add a new command to compile requirements.in to pin
the latest packages in the requirements.txt

### CODE REVIEW

**Comment by ramnes** [comment]
:+1:

File: .travis.yml [hunk 1]
```diff
@@ -48,8 +48,8 @@ before_install:
 
 install:
 # Installl requirements
-  - pip install xcffib
```
Comment (flacjacket) [inline .travis.yml]: We should keep this here to be sure xcffib is always installed before cairocffi.

**Comment by flacjacket** [comment]
I agree that it is good to have a pinned set of packages that we know for sure everything works with, but it would be nice to always test against current releases without having to manually keep the requirements up-to-date.

**Comment by flacjacket** [comment]
I've been thinking about this, and here's my proposal:

Have a separate `requirements` and `requirements-dev`, as you've outlined. The `requirements` will be pinned runtime dependencies (as you've implemented) and will follow the release of the master branch. That way there is a stable release with known dependencies that we know work (I know people mentioned this previously when the cffi breakage was happening). The `requirements-dev` will be what requirements is now, and follow the most recent releases of other packages to make sure we keep qtile working with the most up-to-date libraries.

**Comment by Azd325** [comment]
I like your propsal. I change the code to it.

File: .travis.yml (hunk 1 above)
Comment (Azd325) [inline .travis.yml]: I reverted this change.

**Comment by flacjacket** [comment]
I like this, thanks!

**Pull request merged**